File: apps/web/src/api/capabilities.ts
import { apiGet } from './http'

export type ToolCatalogItem = {
  name: string
  description: string
  tier: string
  unlocked: boolean
}

export type SkillCatalogItem = {
  name: string
  description: string
  tools?: string[]
  mcp?: string[]
}

export type SkillDetail = {
  name: string
  description?: string
  body?: string
  tools?: string[]
  mcp?: string[]
}

export type McpServerItem = {
  name: string
  command?: string
  args?: string[]
  tools?: { name: string; mcp_tool?: string; description?: string }[]
  error?: string
}

export async function fetchTools() {
  return apiGet<{ items: ToolCatalogItem[] }>('/capabilities/tools')
}

export async function fetchSkills() {
  return apiGet<{ items: SkillCatalogItem[] }>('/capabilities/skills')
}

export async function fetchSkillDetail(name: string) {
  return apiGet<{ ok: boolean; item: SkillDetail | null; error?: string }>(
    `/capabilities/skills?name=${encodeURIComponent(name)}`,
  )
}

export async function fetchMcp() {
  return apiGet<{ items: McpServerItem[] }>('/capabilities/mcp')
}

File: backend/apps/api/app/api/v1/capabilities.py
"""能力发现 API：Tool / Skill / MCP 的 L0（及可选 L1）目录。"""

from __future__ import annotations

from fastapi import APIRouter, Depends, Query, Request

from app.core.security import require_api_key

router = APIRouter(prefix="/capabilities")


@router.get("/tools")
async def list_tools(request: Request, _: None = Depends(require_api_key)):
    container = request.app.state.container
    return {"items": container.tools.catalog(unlocked=set())}


@router.get("/skills")
async def list_skills(
    request: Request,
    name: str | None = Query(default=None, description="指定 name 时返回 L1 正文"),
    _: None = Depends(require_api_key),
):
    container = request.app.state.container
    skills = container.skills
    if name:
        result = skills.activate(name)
        if not result.get("ok"):
            return {"ok": False, "error": result.get("error"), "item": None}
        return {
            "ok": True,
            "item": {
                "name": result["name"],
                "description": result.get("description"),
                "body": result.get("body"),
                "tools": result.get("tools"),
                "mcp": result.get("mcp"),
            },
        }
    return {"items": skills.catalog()}


@router.get("/mcp")
async def list_mcp(request: Request, _: None = Depends(require_api_key)):
    container = request.app.state.container
    return {"items": container.tools.mcp.catalog()}
